packages/webapi/state/callview.go

```diff
@@ -44,24 +45,21 @@ func handleCallView(c echo.Context) error {
 	fname := c.Param("fname")
 
 	var params dict.Dict
-	// for some reason c.Bind(&params) doesn't work
 	if c.Request().Body != nil {
 		if err := json.NewDecoder(c.Request().Body).Decode(&params); err != nil {
 			return httperrors.BadRequest("Invalid request body")
 		}
 	}
-
 	theChain := chains.AllChains().Get(chainID)
 	if theChain == nil {
 		return httperrors.NotFound(fmt.Sprintf("Chain not found: %s", chainID))
 	}
-
-	vctx, err := viewcontext.NewFromDB(database.GetKVStore(theChain.ID()), *theChain.ID(), theChain.Processors())
-	if err != nil {
-		return fmt.Errorf(fmt.Sprintf("Failed to create context: %v", err))
-	}
-
-	ret, err := vctx.CallView(contractHname, coretypes.Hn(fname), params)
+	vctx := viewcontext.NewFromChain(theChain)
+	var ret dict.Dict
+	_ = optimism.RepeatOnceIfUnlucky(func() error {
```

Comment:
should we handle the error here in case `CallView` errors twice?
```suggestion
	err = optimism.RepeatOnceIfUnlucky(func() error {
```

Reply:
the error is assigned to the `err` in the closure. So, it is handled

Reply:
the return error from `RepeatOnceIfUnlucky` may be useful when we use closure without inline context